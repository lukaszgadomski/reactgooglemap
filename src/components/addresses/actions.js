import actionTypes from "./actionTypes";

export const clickAddress = address => {
  return {
    type: actionTypes.CLICK_ADDRESS,
    address
  };
};

export const fetchAddressRequest = () => {
  return {
    type: actionTypes.FETCH_REQUEST_ADDRESS,
    fetching: true
  };
};

export const fetchAddressError = errMsg => {
  return {
    type: actionTypes.FETCH_ERROR_ADDRESS,
    fetching: false,
    err: errMsg
  };
};

export const fetchAddressSuccess = res => {
  return {
    type: actionTypes.FETCH_SUCCESS_ADDRESS,
    fetching: false,
    addresses: res.addresses,
    markers: res.markers
  };
};
