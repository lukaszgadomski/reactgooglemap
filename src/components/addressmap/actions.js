import actionTypes from "./actionTypes";

export const clickAddress = address => {
  return {
    type: actionTypes.CLICK_ADDRESS,
    address
  };
};
