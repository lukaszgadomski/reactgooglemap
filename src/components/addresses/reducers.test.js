import reducer from "./reducers";
import types from "./ActionTypes";
import initialState from "../../storeInitialState";
import * as actions from "./actions";

describe("todos reducer", () => {
  test("should handle CLICK_ADDRESS", () => {
    const givenSelected = {
      id: 1,
      address: "Warszawa, ul. Grzybowska 37",
      hours: [
        { day: "Niedziela", hours: "brak danych" },
        { day: "Poniedziałek", hours: "10:00 - 19:00" },
        { day: "Wtorek", hours: "10:00 - 19:00" },
        { day: "Środa", hours: "10:00 - 19:00" },
        { day: "Czwartek", hours: "10:00 - 19:00" },
        { day: "Piątek", hours: "10:00 - 19:00" },
        { day: "Sobota", hours: "brak danych" }
      ],
      name: "Alsen",
      lat: "52.2351646",
      lng: "20.9942039"
    };
    expect(
      reducer(initialState.addresses, actions.clickAddress(givenSelected))
    ).toMatchSnapshot();
  });
});
