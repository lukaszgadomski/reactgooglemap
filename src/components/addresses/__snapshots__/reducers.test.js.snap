// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos reducer should handle CLICK_ADDRESS 1`] = `
Object {
  "list": Array [],
  "selected": Object {
    "address": "Warszawa, ul. Grzybowska 37",
    "hours": Array [
      Object {
        "day": "Niedziela",
        "hours": "brak danych",
      },
      Object {
        "day": "Poniedziałek",
        "hours": "10:00 - 19:00",
      },
      Object {
        "day": "Wtorek",
        "hours": "10:00 - 19:00",
      },
      Object {
        "day": "Środa",
        "hours": "10:00 - 19:00",
      },
      Object {
        "day": "Czwartek",
        "hours": "10:00 - 19:00",
      },
      Object {
        "day": "Piątek",
        "hours": "10:00 - 19:00",
      },
      Object {
        "day": "Sobota",
        "hours": "brak danych",
      },
    ],
    "id": 1,
    "lat": "52.2351646",
    "lng": "20.9942039",
    "name": "Alsen",
  },
}
`;
